### DISCUSSION TITLE
Linux Terminal Plugin

### DISCUSSION THREAD

**Discussion by datenwurm** [zim-desktop-wiki/zim-desktop-wiki#2488] [Ideas]
Hello everyone,

A while back, I developed a [Linux Terminal Plugin](https://github.com/datenwurm/zim-plugin-terminal) for Zim. Since i'm using it quite a lot I'm wondering if others in the community might find it beneficial as well.

Do you think this plugin might be a good addition to the main branch? Looking forward to hearing your thoughts!

Best,
Thomas

**Comment by introt** [comment]
Hi, big fan of the plugin here :)

On one hand, I'd like to see it "mainlined" - if only to force the Flatpak to support it as well[^1]. We already have Linux-only plugins like the Zeitgeist logger.

On the other hand, it could be a great fit to the "third-party plugin store", previously discussed over at #1936 - it's self-contained, and could also demonstrate how to package a Zim plugin as a Flatpak plugin - ~after someone figures out how all that works we could stop shipping our own copy of LaTeX in the Flatpak.~ my bad, it's already an extension: https://github.com/flathub/org.zim_wiki.Zim/blob/master/org.zim_wiki.Zim.yaml

How much time would you be interested to contribute? ;)

[^1]: `vte` is required: https://github.com/flathub/org.gnome.Builder/blob/3c18019bc5666e049c3525489e25e83cd6a128df/org.gnome.Builder.json#L274-L290

**Reply by datenwurm** [reply]
@introt: Sorry for the late reply! Really appreciate your thoughtful feedback about the Flatpak integration. While I can't commit time to this currently, I'm definitely keen to explore these ideas when my schedule opens up.

**Reply by introt** [reply]
No worries! I feel you with the busy schedule, make sure to take time for yourself every now and then. Take care and happy hacking!

**Comment by nh3oh** [comment]
Hi!

I just discover this plugin...

@datenwurm   Is there a way to change the current path color for a lighter blue achieving more contrast on dark background?

Thanks for sharing!

Best regards.

**Reply by datenwurm** [reply]
@nh30h: i think the color of the current path is handled by PS1 (e.g. in ~/.bashrc)

**~/.bashrc:**
```
# make path light blue
export PS1='\[\e[38;2;100;149;237m\]\w\[\e[0m\]$ '
```

If you want to go fancy, you may also wanna try https://github.com/b-ryan/powerline-shell.

**Comment by nh3oh** [comment]
Thanks for quick hint!

As changing `$PS1` in `~/.bashrc` (yet configured for dark background) doesn't fix path color in the plugin and taping the export every time would be hassle,
I've placed an alias in `~/.bash_aliases` :
```
alias zimterm='export PS1="\[\e]0;\u@\h: \w\a\]${debian_chroot:+($debian_chroot)}\[\033[01;32m\]\u@\h\[\033[00m\]:\[\033[38;2;100;149;237m\]\w\[\033[00m\]\$ "'
```

This way is not best one which may be custom hard coding  the plugin... if possible (I've no idea about).

---------
**Added:**  _I guess I'm messing with `PS1` sorry for that..._

**Reply by datenwurm** [reply]
@nh3oh: I've added support for custom startup commands in the terminal plugin. You can now set an initialization command in the preferences dialog:

![image](https://github.com/user-attachments/assets/773d3056-152e-4567-860d-e116cf714ca0)

For example, to customize your prompt, you could use:
```
export PROMPT_COMMAND=''; export PS1='test'
``` 
Hope this helps! Let me know if you have any questions or run into issues!

**Reply by introt** [reply]
Nice! Could you also add a on-by-default checkbox that sets a `TERM_IS_ZIM=1` environment variable when initializing so one could check for that in their `.bashrc` and/or other scripts?

Given our single-main-window-per-wiki behavior, setting a `ZIM_WIKI=shorthand` could also be useful context for some scripts - as could `PATH_TO_CURRENT_ZIM_PAGE` as an alternative to "Automatically switch to new path on page change".

**Comment by nh3oh** [comment]
Yes, it does... Nice shot !

Thanks a lot.
